frontend/components/Blocks/HeroBlock/HeroBlock.tsx

```diff
@@ -0,0 +1,117 @@
+import { useState, useEffect } from "react";
+import Image from "next/image";
+import ReactPlayer from "react-player/lazy";
+import { ArrowDownIcon } from "@heroicons/react/24/outline";
+
+import RawHtml from "../../RawHtml";
+
+type Props = {
+  data: {
+    type: string;
+    value: {
+      backgroundColor: string;
+      title: string;
+      text: string;
+      media: [];
+      alt_text: string;
+      button?: {
+        button_style: string;
+        button_text: string;
+        button_hyperlink: string;
+      };
+    };
+    id: string;
+  };
+};
+
+export default function HeroBlock({ data }: Props) {
+  const [imageSize, setImageSize] = useState({
+    width: 1,
+    height: 1,
+  });
+  const [hasWindow, setHasWindow] = useState(false);
+
+  const backgroundcolor = data.value.backgroundColor;
+  const mediaDetails = data.value.media[0];
+
+  useEffect(() => {
+    if (typeof window !== "undefined") {
+      setHasWindow(true);
+    }
+  }, []);
```

Comment:
```suggestion
  const backgroundcolor = data.value.backgroundColor;
  const mediaDetails = data.value.media[0];

  const hasWindow = typeof window !== "undefined"
```

Reply:
This one I had to leave in for now. If I take the setting of window out of the useEffect then I get a hydration error with the ReactPlayer. This was the solution I found online and I haven't found another way of fixing this. Any suggestions would be welcome though.